dll/win32/shell32/wine/brsfolder.c

```diff
@@ -122,12 +122,23 @@ static inline DWORD BrowseFlagsToSHCONTF(UINT ulFlags)
     return SHCONTF_FOLDERS | (ulFlags & BIF_BROWSEINCLUDEFILES ? SHCONTF_NONFOLDERS : 0);
 }
 
+#ifndef __REACTOS__
 static void browsefolder_callback( LPBROWSEINFOW lpBrowseInfo, HWND hWnd,
                                    UINT msg, LPARAM param )
+#else
+static BOOL browsefolder_callback( LPBROWSEINFOW lpBrowseInfo, HWND hWnd,
+                                   UINT msg, LPARAM param )
+#endif
 {
+#ifndef __REACTOS__
     if (!lpBrowseInfo->lpfn)
         return;
     lpBrowseInfo->lpfn( hWnd, msg, param, lpBrowseInfo->lParam );
+#else
+    if (!lpBrowseInfo->lpfn)
+        return FALSE;
+    return lpBrowseInfo->lpfn( hWnd, msg, param, lpBrowseInfo->lParam );
+#endif
```

Comment:
I was looking through MSDN documentation and I'm not sure this is correct. The documentation says this:
> Returns zero except in the case of **BFFM_VALIDATEFAILED**. For that flag, returns zero to dismiss the dialog or nonzero to keep the dialog displayed.

https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb762598(v=vs.85)

Reply:
The callback function you refer is implemented in newdev/wizard.c 
BrowseCallbackProc(HWND   , UINT   ,   LPARAM ,   LPARAM). 
This function have diferent parameters browsefolder_callback (LPBROWSEINFOW , HWND,  UINT , LPARAM).